File: Backend/src/main/java/com/example/Java_Backend/entities/Order.java
package com.example.Java_Backend.entities;

import jakarta.persistence.*;
import lombok.Data;
import com.example.Java_Backend.entities.OrderDetail;


import java.time.DateTimeException;
import java.time.LocalDateTime;
import java.util.List;

@Data
public class Order {
    private Integer Id;

    private String UserId;

    private LocalDateTime OrderDate;

    private Double TotalAmount;

    private String Status;

    private String ReceiverName;

    private String ReceiverPhone;

    private String ShippingProvince;

    private String ShippingDistrict;

    private String ShippingWard;

    private String ShippingDetailAddress;

    private String ShippingNote;

    private String PaymentMethod;

    private String VoucherCode;

    private Double DiscountAmount;

    private Double FinalAmount;


    private List<OrderDetail> OrderDetails;
}

File: KHE_AuthService/KHE_AuthService/Entities/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace KHE_AuthService.Entities
{
    public class Order
    {
        public int Id { get; set; }

        public string UserId { get; set; } = null!;

        public DateTime OrderDate { get; set; }

        public decimal TotalAmount { get; set; }

        public string Status { get; set; } = null!;

     
        public string? ReceiverName { get; set; }

        public string? ReceiverPhone { get; set; }
        public string? ReceiverEmail { get; set; }

      
        public string? ShippingProvince { get; set; }

        public string? ShippingDistrict { get; set; }

        public string? ShippingWard { get; set; }

        public string? ShippingDetailAddress { get; set; }

        public string? ShippingNote { get; set; }

        public string? PaymentMethod { get; set; }

        public virtual ICollection<OrderDetail> OrderDetails { get; set; } = new List<OrderDetail>();

        public virtual User User { get; set; } = null!;
    }
}
